app_dart/lib/src/request_handling/swarming_authentication.dart

```diff
@@ -111,9 +100,57 @@ class SwarmingAuthenticationProvider extends AuthenticationProvider {
 
       return AuthenticatedContext(agent: agent, clientContext: clientContext);
     } else if (swarmingToken != null) {
-      return await authenticateIdToken(swarmingToken, clientContext: clientContext, log: log);
+      return await authenticateAccessToken(swarmingToken, clientContext: clientContext, log: log);
     }
 
     throw const Unauthenticated('Request rejected due to not from LUCI or Cocoon agent');
   }
+
+  /// Authenticate [accessToken] against Google OAuth 2 API.
+  ///
+  /// Access tokens are the legacy authentication strategy for Google OAuth, where ID tokens
+  /// are the new technique to use. LUCI auth only generates access tokens, and must be
+  /// validated against a different endpoint. We only validate access tokens if they belong
+  /// to a DeviceLab service account.
+  ///
+  /// If LUCI auth adds id tokens, we can switch to that and remove this.
+  Future<AuthenticatedContext> authenticateAccessToken(String accessToken,
+      {ClientContext clientContext, Logging log}) async {
+    // Authenticate as a signed-in Google account via OAuth id token.
+    final HttpClient client = httpClientProvider();
+    try {
+      final HttpClientRequest verifyTokenRequest = await client.getUrl(Uri.https(
```

Comment:
Using the Google OAuth apis was malforming the access token request, causing a 400 bad request to return from the API. I scratched it and went for this approach as it's similar to id tokens.

Reply:
Is this an error with the APIs or the client libraries making the request?

Reply:
The client libraries as the the request looks to be putting all possible query parameters in. I'm not sure who owns that library, but to unblock for now I went with this approach.

Reply:
Actually the question was to decide if we need to open a bug for the client library but not a blocker for this PR.

Reply:
I'll create a reproduction and file an issue.